BYOND 510

neersighted opened the pull request:
# BYOND 510 Beta

510 beta is [here](http://www.byond.com/download/build/510/)! HYPE!

This branch uses new 510 features, and stubs them with softcoded versions when running on older versions. These stubs are found in `code/__HELPERS/510.dm`.

Note that this rewrites fullscreen overlay code in order to function on both 509 and 510 as expected.

neersighted committed:
Get the code running on 510

* Travis for 510
* Remove json, list2text, text2list, bygex
* Change blind and click catcher to a low plane

nullquery commented:
Does this fix everything 510-wise? I need 510 for the json_encode and json_decode procs.

Would probably be better to add ifdef's to check the compiler version.

e.x.

#if DM_VERSION < 510
OLD CODE
#else
NEW CODE
#endif

Then this can be merged into the master branch w/o problems. :+1:
The DM_VERSION check can be removed later, when 510 is stable.

Supermichael777 commented:
if lizards are a maintainability issue they could be cut(this is a joke don't do this)

Cheridan commented:
Yeah Travis is erroring out on map files randomly. @tgstation/commit-access Just do a manual checkout and test compilation.

neersighted committed:
Stub out 510 API

Also update travis to 1321

neersighted committed:
Re-implement lizard hissing

Note that this reverts #13656, due to this bug (http://www.byond.com/forum/?post=2027070)

neersighted committed:
Re-implement screen overlays

This fixes 510 showing all overlays in the rightclick menu

tkdrg commented:
just writing this down but
* you forgot to redo the notes stuff
* you forgot to redo the lizard sss, just stub regex with the thing ginja mentioned in the report but do it outside the 509 ifdef

MrStonedOne commented on code/__HELPERS/_string_lists.dm:
```diff
@@ -12,11 +12,11 @@ var/global/list/string_cache
 			fileList = file2list("strings/[filename]")
 			for(var/s in fileList)
-				stringsList = text2list(s, "@=")
+				stringsList = splittext(s, "@=")
```
So, i'm just gonna point out, that this doesn't actually work.

luckly it will most likely work in 510.1322

http://www.byond.com/forum/?post=2026759&page=2

neersighted committed:
Properly layer and set/clear fullscreen overlays

This fixes ordering issues and admin ghosting

neersighted committed:
Stub regex.Replace, re-implement SQL note conversion

neersighted committed:
Support Regex in replaceall, re-implement lizard hiss limits

neersighted commented:
This does not currently work because of what MSO pointed out, as well as a bug in regex. 1322 should resolve all this for us.

MrPerson commented:
Please follow CONTRIBUTING.md, especially this part near the bottom:

>If you are proposing multiple changes, which change many different aspects of the code, you are expected to section them off into different pull requests in order to make it easier to review them and to deny/accept the changes that are deemed acceptable.

optimumtact commented on code/_onclick/click.dm:
```diff
@@ -331,7 +331,7 @@
 	icon = 'icons/mob/screen_full.dmi'
 	icon_state = "passage0"
-	layer = 0
+	plane = -99
```
Define this

optimumtact commented on code/modules/mob/living/silicon/silicon.dm:
```diff
@@ -135,9 +135,10 @@
 		if(2)
 			src.take_organ_damage(10)
-	flick("noise", src:flash)
+			Stun(3)
```
You've added a stun that wasn't here before?

optimumtact commented on code/__HELPERS/510.dm:
```diff
@@ -0,0 +1,253 @@
+// Helpers for running 510 code on older versions of BYOND.
+#if DM_VERSION < 510
+#define BYGEX "code/__HELPERS/bygex.dll"
```
this is going to break the linux instructions I have because you're now directing to the exact path instead of the inexact one previously

optimumtact commented on code/modules/tgui/tgui.dm:
```diff
@@ -245,10 +245,11 @@
 	// Generate the JSON.
-	var/json = JSON.stringify(json_data)
+	var/json = json_encode(json_data)
+	//var/json = JSON.stringify(json_data)
```
no need to leave these in

neersighted committed:
Remove eroneous code; click catcher define; bygex path fix

duncathan commented on README.md:
```diff
@@ -73,11 +73,6 @@ specified in the config.txt, and set the Security box to 'Safe'.  Then press GO
-We use BYGEX for some of our text replacement related code. Unfortunately, we
-only have a windows dll included right now. You can find a version known to compile on linux, along with some basic install instructions here
-https://github.com/optimumtact/byond-regex
-
```
Might want to mention that BYGEX is used for legacy purposes

neersighted replied:
I think Linux users would be better off running the 519 beta, which makes this a non issue for them.

nullquery replied:
519?!

RemieRichards replied:
@nullquery oh my gosh I hate you.
I just saw the email and was about to make my own 519 joke ;_;

neersighted replied:
510. Damn fat fingers.

neersighted committed:
Fix lizard hissing

tkdrg commented on .travis.yml:
```diff
@@ -3,8 +3,8 @@ sudo: false
-    - BYOND_MAJOR="509"
-    - BYOND_MINOR="1319"
+    - BYOND_MAJOR="510"
+    - BYOND_MINOR="1321"
```
1322

tkdrg commented on code/_onclick/hud/fullscreen.dm:
```diff
@@ -0,0 +1,107 @@
+
+	if(animate)
+		animate(screen, alpha = 0, time = animate)
+		sleep(animate)
```
Are you sure this is safe? this can lock up controllers etc if called carelessly. it might make more sense to spawn() something off here instead.

MrPerson replied:
set waitfor = 0, no spawn

neersighted replied:
I was told that life() is set waitfor 0.

tkdrg commented on code/game/objects/items/devices/flashlight.dm:
```diff
@@ -80,8 +79,7 @@
 			else	//they're okay!
-				if(!M.eye_blind)
-					flick("flash", M.flash)	//flash the affected mob
+				if(M.flash_eyes())
```
why are you turning penlights into ghetto flashes

MrPerson replied:
If you want, add a visual-only or nostun arg to the proc

neersighted committed:
Fix flashlights actually stunning

Also make clear_fullscreen() waitfor = 0 to be safe,
and bump travis version
